src/bfcli/dump.h

```diff
@@ -0,0 +1,27 @@
+
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * Copyright (c) 2023 Meta Platforms, Inc. and affiliates.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+
+#include "core/marsh.h"
+
+/**
+ * @file dump.h
+ *
+ * Tools for printing CLI data to the console.
+ */
+
+/**
+ * Print ruleset information and counters to the console.
+ *
+ * @param chains_and_counters_marsh Pointer to the marshalled chains and counters returned by the daemon.
```

Comment:
No marsh should be used here. It should be converted back to a list of chains directly from libbpfilter.